### ISSUE TITLE
Needs an update as of Sep 6

### ISSUE THREAD

**Issue by LilyBageL** [morgana-x/gmod-chatgpt-npcs#2]
The mod does give the NPCs random names, and console does show inputs and outputs (albeit quite broken and usually just repeats "this is an example of how a response should be"). But NPCs do not respond with any readable sentences or even just any words. I do have a list of mods but they should have no files conflicting with an OpenAI-integration mod. This add-on carries a really outstanding concept of idea, and I was wondering if you're planning to update this potential masterpiece but simply forgotten, or is currently occupied with different projects?

TL;DR: Mod's not working, wanted to know if you're planning to update it or not

**Comment by morgana-x** [comment]
If I were to update this mod I would switch to character.ai(Using unofficial stuff) as I'm pretty sure OpenAI's ChatGPT api no longer has free trials, right now I'm doing other things but I might update it sometime in the future when I have time

And yeah I would probably completely rewrite it if/when I update it since the code is atrocious

**Comment by LilyBageL** [comment]
Sorry for the late notice, even though I'm less active on gmod now, it'd still be great to see your concept and idea work in its peak performance, AI-generated NPC interactions in gmod would be extremely hilarious and most definitely popular with content creators.

AI Roguelite and AI Roguelite 2D by Max Loh should be extremely inspiring for you to check them out, I played those two games, they're on steam. Max Loh incorporated his own kind of free cloud service for text generation, but for audio and voice on AI Roguelite 2D, he incorporates third-party services that you might also be able to use in your project.

**Comment by morgana-x** [comment]
Recently I have been running Large Language Models and Deep learning speech synthesis programs locally on my computer to make chat assistants, I'd probably use something like that but hosting services is probably out of both my wallet and hardware's capability, but maybe I could make a version of this for people with good computers?

**Comment by morgana-x** [comment]
Should say I don't have too much interest in gmod anymore either and am nearing the end of my high school studies (exams etc)

**Comment by LilyBageL** [comment]
Obviously focus on real life first.

Hosting locally also works, I like AI Roguelite's user friendly interface when it comes to that, you just copy the link to a model on huggingface, paste it in the model box and the game automatically do all the downloading and setting up for you.

Although, the code in AI Roguelite seems to be worse than 2D, as everything runs slower than the mentioned successor. So if you want to take inspiration from the games, best to consult the author of both games instead of digging around code yourself. Obviously if you know how to code it in already it's great and amazing.
